changelog/17160.txt

```diff
@@ -0,0 +1,3 @@
+```release-note:improvement
+auth/gcp: bump github.com/hashicorp/vault-plugin-auth-gcp to v0.14.0
```

Comment:
I haven't seen us adding changelog entries for the plugin bumps themselves. The entries have mostly been focused on the bug fixes, improvements, and features added to plugins. I'm thinking this CL entry should include an entry similar to https://github.com/hashicorp/vault/pull/17174/files#diff-7c432b8c3bb8887f5e0b389cbb81f2bdf4a6964f0a42dc395a6ed5aaf639c7fc?

Reply:
That makes sense, I opened a PR to update the previous changelog here: https://github.com/hashicorp/vault/pull/17176